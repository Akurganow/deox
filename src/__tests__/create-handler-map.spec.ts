import { createAction } from '../create-action'
import { createHandlerMap as handle } from '../create-handler-map'

describe('createHandlerMap', () => {
  it('should belong one action to one handler', () => {
    const increment = createAction('[Counter] increment')

    expect(handle(increment, (state: number) => state + 1)).toMatchSnapshot()
  })

  it('should belong multiple actions to one handler', () => {
    const increment = createAction('[Counter] increment')
    const increase = createAction('[Counter] increase')

    expect(
      handle([increment, increase], (state: number) => state + 1)
    ).toMatchSnapshot()
  })
})
