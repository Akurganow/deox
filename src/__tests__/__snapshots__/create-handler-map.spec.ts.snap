// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createHandlerMap should belong multiple actions to one handler 1`] = `
Object {
  "[Counter] increase": [Function],
  "[Counter] increment": [Function],
}
`;

exports[`createHandlerMap should belong one action to one handler 1`] = `
Object {
  "[Counter] increment": [Function],
}
`;
